### PR TITLE
ARTEMIS-1595 Fixing serialization issues between Artemis 1.5 and master

### PR DESCRIPTION
**Pull request by clebertsuconic**

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ProtocolHandler.java [hunk 1]
```diff
@@ -143,6 +143,8 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
          }
       }
 
+
+      // the innitial communication happens here...
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ProtocolHandler.java]: hahaha you left this in from the web conf :)

File: artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnectionFactory.java [hunk 2]
```diff
@@ -95,6 +95,7 @@ public void writeExternal(ObjectOutput out) throws IOException {
       URI uri = toURI();
 
       try {
+         System.out.println("writing " + uri.toASCIIString());
```
Comment (jmesnil) [inline artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnectionFactory.java]: should be removed

File: artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnectionFactory.java [hunk 3]
```diff
@@ -188,13 +189,15 @@ public void setDeserializationWhiteList(String whiteList) {
    @Override
    public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
       String url = in.readUTF();
+      System.out.println("Reading " + url);
```
Comment (jmesnil) [inline artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnectionFactory.java]: should be removed

File: artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnectionFactory.java [hunk 4]
```diff
@@ -188,13 +189,15 @@ public void setDeserializationWhiteList(String whiteList) {
...
       ConnectionFactoryParser parser = new ConnectionFactoryParser();
       ServerLocatorParser locatorParser = new ServerLocatorParser();
       try {
          URI uri = new URI(url);
          serverLocator = locatorParser.newObject(uri, null);
          parser.populateObject(uri, this);
       } catch (Exception e) {
+         e.printStackTrace();
```
Comment (jmesnil) [inline artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnectionFactory.java]: should be removed

**Comment by clebertsuconic** [comment]
Forgot those. Will remove them. 


Thanks for catching this.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ProtocolHandler.java (hunk 1 above)
Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ProtocolHandler.java]: ouch.. I will separate this commit and keep it :)
Live coding!!!

**Comment by clebertsuconic** [comment]
@jmesnil all fixed

**Comment by michaelandrepearce** [comment]
+1 will merge

**Pull request closed without merging**